bcda/api/requests.go

```diff
@@ -156,9 +164,16 @@ func (h *Handler) bulkRequest(resourceTypes []string, w http.ResponseWriter, r *
 	// Overall, this will prevent a queue of concurrent calls from slowing up our system.
 	// NOTE: this logic is relevant to PROD only; simultaneous requests in our lower environments is acceptable (i.e., shared opensbx creds)
 	if (os.Getenv("DEPLOYMENT_TARGET") == "prod") && (!db.Find(&jobs, "aco_id = ?", acoID).RecordNotFound()) {
-		if types, ok := check429(jobs, resourceTypes, w); !ok {
-			w.Header().Set("Retry-After", strconv.Itoa(utils.GetEnvInt("CLIENT_RETRY_AFTER_IN_SECONDS", 0)))
-			w.WriteHeader(http.StatusTooManyRequests)
+		if types, err := check429(jobs, resourceTypes, version); err != nil {
```

Comment:
Using a custom error type instead of the boolean to avoid multiple writes the Http.ResponseWriter.

If we encountered a non duplicate error (i.e. url parsing issue), we would need to do an early return AFTER we've written an operational outcome with status code.

By returning a custom error type, we can handle the response writing in the bulkRequest function and do not have to pass in the writer.